Origin: EmberBSD (AI-assisted); not submitted upstream.
GCC 16.2.0 with Iain Sandoe's Darwin port and the pkgsrc patches.
Add the darwin_xpc_choose fixincludes hack for macOS <xpc/base.h>.
XPC_ATTRIBUTE pastes __has_attribute onto _XPC_CHOOSE_, but the header
defines only _XPC_CHOOSE_0 and _XPC_CHOOSE_1.  GCC reports the standard
version for deprecated and noreturn (202311 in C, 200809 and 201309 in
C++), so any GCC translation unit that includes <xpc/xpc.h> fails.
fixincl.x is regenerated with AutoGen 5.18.16 from the patched inclhack.def.
GCC source retains its GPLv3+ licence and upstream copyright notices.

--- fixincludes/inclhack.def.orig
+++ fixincludes/inclhack.def
@@ -1762,6 +1762,27 @@
 };
 
 /*
+ *  macOS <xpc/base.h> pastes the value of __has_attribute onto _XPC_CHOOSE_
+ *  and defines only _XPC_CHOOSE_0 and _XPC_CHOOSE_1.  GCC returns the
+ *  standard attribute version for deprecated and noreturn (202311 in C,
+ *  200809 and 201309 in C++), which leaves an undefined macro call.
+ */
+fix = {
+    hackname  = darwin_xpc_choose;
+    mach      = "*-*-darwin*";
+    files     = xpc/base.h;
+    select    = "^#define _XPC_CHOOSE_1\\(\\.\\.\\.\\) __VA_ARGS__$";
+    bypass    = "_XPC_CHOOSE_202311";
+    c_fix     = format;
+    c_fix_arg = "%0\n"
+		"#define _XPC_CHOOSE_200809(...) __VA_ARGS__\n"
+		"#define _XPC_CHOOSE_201309(...) __VA_ARGS__\n"
+		"#define _XPC_CHOOSE_202311(...) __VA_ARGS__";
+    test_text = "#define _XPC_CHOOSE_0(...)\n"
+		"#define _XPC_CHOOSE_1(...) __VA_ARGS__";
+};
+
+/*
  *  Fix <c_asm.h> on Digital UNIX V4.0:
  *  It contains a prototype for a DEC C internal asm() function,
  *  clashing with gcc's asm keyword.  So protect this with __DECC.
--- fixincludes/fixincl.x.orig
+++ fixincludes/fixincl.x
@@ -2,11 +2,11 @@
  *
  * DO NOT EDIT THIS FILE   (fixincl.x)
  *
- * It has been AutoGen-ed  February 16, 2026 at 09:10:33 AM by AutoGen 5.18.16
+ * It has been AutoGen-ed  October 24, 2026 at 02:42:39 PM by AutoGen 5.18.16
  * From the definitions    inclhack.def
  * and the template file   fixincl
  */
-/* DO NOT MERGE THIS FILE, EITHER Mon Feb 16 09:10:33 GMT 2026
+/* DO NOT MERGE THIS FILE, EITHER Sat Oct 24 14:42:39 UTC 2026
  *
  * You must regenerate it.  Use the ./genfixes script.
  *
@@ -15,7 +15,7 @@
  * certain ANSI-incompatible system header files which are fixed to work
  * correctly with ANSI C and placed in a directory that GNU C will search.
  *
- * This file contains 249 fixup descriptions.
+ * This file contains 250 fixup descriptions.
  *
  * See README for more information.
  *
@@ -3598,6 +3598,53 @@
 
 /* * * * * * * * * * * * * * * * * * * * * * * * * *
  *
+ *  Description of Darwin_Xpc_Choose fix
+ */
+tSCC zDarwin_Xpc_ChooseName[] =
+     "darwin_xpc_choose";
+
+/*
+ *  File name selection pattern
+ */
+tSCC zDarwin_Xpc_ChooseList[] =
+  "xpc/base.h\0";
+/*
+ *  Machine/OS name selection pattern
+ */
+tSCC* apzDarwin_Xpc_ChooseMachs[] = {
+        "*-*-darwin*",
+        (const char*)NULL };
+
+/*
+ *  content selection pattern - do fix if pattern found
+ */
+tSCC zDarwin_Xpc_ChooseSelect0[] =
+       "^#define _XPC_CHOOSE_1\\(\\.\\.\\.\\) __VA_ARGS__$";
+
+/*
+ *  content bypass pattern - skip fix if pattern found
+ */
+tSCC zDarwin_Xpc_ChooseBypass0[] =
+       "_XPC_CHOOSE_202311";
+
+#define    DARWIN_XPC_CHOOSE_TEST_CT  2
+static tTestDesc aDarwin_Xpc_ChooseTests[] = {
+  { TT_NEGREP,   zDarwin_Xpc_ChooseBypass0, (regex_t*)NULL },
+  { TT_EGREP,    zDarwin_Xpc_ChooseSelect0, (regex_t*)NULL }, };
+
+/*
+ *  Fix Command Arguments for Darwin_Xpc_Choose
+ */
+static const char* apzDarwin_Xpc_ChoosePatch[] = {
+    "format",
+    "%0\n\
+#define _XPC_CHOOSE_200809(...) __VA_ARGS__\n\
+#define _XPC_CHOOSE_201309(...) __VA_ARGS__\n\
+#define _XPC_CHOOSE_202311(...) __VA_ARGS__",
+    (char*)NULL };
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * *
+ *
  *  Description of Dec_Intern_Asm fix
  */
 tSCC zDec_Intern_AsmName[] =
@@ -10188,9 +10235,9 @@
  *
  *  List of all fixes
  */
-#define REGEX_COUNT          288
+#define REGEX_COUNT          290
 #define MACH_LIST_SIZE_LIMIT 187
-#define FIX_COUNT            249
+#define FIX_COUNT            250
 
 /*
  *  Enumerate the fixes
@@ -10280,6 +10327,7 @@
     DARWIN_STDINT_7_FIXIDX,
     DARWIN_UCRED__ATOMIC_FIXIDX,
     DARWIN_FLT_EVAL_METHOD_FIXIDX,
+    DARWIN_XPC_CHOOSE_FIXIDX,
     DEC_INTERN_ASM_FIXIDX,
     DJGPP_WCHAR_H_FIXIDX,
     ECD_CURSOR_FIXIDX,
@@ -10868,6 +10916,11 @@
      DARWIN_FLT_EVAL_METHOD_TEST_CT, FD_MACH_ONLY | FD_SUBROUTINE,
      aDarwin_Flt_Eval_MethodTests,   apzDarwin_Flt_Eval_MethodPatch, 0 },
 
+  {  zDarwin_Xpc_ChooseName,    zDarwin_Xpc_ChooseList,
+     apzDarwin_Xpc_ChooseMachs,
+     DARWIN_XPC_CHOOSE_TEST_CT, FD_MACH_ONLY | FD_SUBROUTINE,
+     aDarwin_Xpc_ChooseTests,   apzDarwin_Xpc_ChoosePatch, 0 },
+
   {  zDec_Intern_AsmName,    zDec_Intern_AsmList,
      apzDec_Intern_AsmMachs,
      DEC_INTERN_ASM_TEST_CT, FD_MACH_ONLY,
--- fixincludes/tests/base/xpc/base.h.orig
+++ fixincludes/tests/base/xpc/base.h
@@ -0,0 +1,18 @@
+/*  DO NOT EDIT THIS FILE.
+
+    It has been auto-edited by fixincludes from:
+
+	"fixinc/tests/inc/xpc/base.h"
+
+    This had to be done to correct non-standard usages in the
+    original, manufacturer supplied header file.  */
+
+
+
+#if defined( DARWIN_XPC_CHOOSE_CHECK )
+#define _XPC_CHOOSE_0(...)
+#define _XPC_CHOOSE_1(...) __VA_ARGS__
+#define _XPC_CHOOSE_200809(...) __VA_ARGS__
+#define _XPC_CHOOSE_201309(...) __VA_ARGS__
+#define _XPC_CHOOSE_202311(...) __VA_ARGS__
+#endif  /* DARWIN_XPC_CHOOSE_CHECK */
